### PR TITLE
C++: Don't generate `Load`s for constant expressions

### PR DESCRIPTION
**Pull request by MathiasVP**

### CODE REVIEW

File: cpp/ql/test/library-tests/ir/modulus-analysis/test.cpp [hunk 1]
```diff
@@ -19,13 +19,11 @@ void m(int i, bool cond, int x, int y) {
     mod(j); // $ mod=0,3,4
 
     if (x % c1 == 3 && y % c1 == 7) {
-        // Need implies_v2
```
Comment (MathiasVP) [inline cpp/ql/test/library-tests/ir/modulus-analysis/test.cpp:22]: Turns out this didn't actually need `implies_v2` 😂. Just proper IR!

**Review by andersfugmann — APPROVED** [review]
LGTM.

**Comment by andersfugmann** [comment]
DCA query results looks good; solved 6 FN's and 2 TP's. 

I think this also reduced IR inconsistencies on `operandAcrossFunctions` and `nonUniqueEnclosingIRFunction`, but introduced two more `unnecessaryPhiInstructions`  on `openzfs`.

One experiment failed, but is seems to be an extraction failure. 

@MathiasVP Is the changes for unnecessaryPhiInstructions ir concistency check on openzfs something to be concerned about?

**Comment by MathiasVP** [comment]
Indeed the DCA results loon great 🎉.

I don't think the new inconsistencies is something to worry about. I'll see what they are just to be sure, though 👀

**Comment by MathiasVP** [comment]
> I don't think the new inconsistencies is something to worry about. I'll see what they are just to be sure, though 👀

Nothing to blame on this PR. We're simply just generating different IR for a function that's duplicated based on what looks like an ODR violation.

For those interested: The difference in IR is because we know realize that [this branch](https://github.com/openzfs/zfs/blob/8469b5aac0cee4f0e8b13018c3e83129554a6945/module/zfs/dmu_objset.c#L2814) is always true (because `ntasks` is equal to `dmu_find_threads`, and because of this PR we now correctly model the value of this constant integer and realize that it's always 0).

**Pull request merged**